diff -ur common/BuildSettings.h.bak common/BuildSettings.h
--- common/BuildSettings.h.bak	2018-09-15 15:07:34.000000000 +0200
+++ common/BuildSettings.h	2019-08-24 20:33:13.718750000 +0200
@@ -731,6 +731,20 @@
 
 
 
+// standard library quirks
+
+#if MPT_CXX_AT_LEAST(17)
+#if (MPT_COMPILER_GCC || MPT_COMPILER_CLANG)
+// we need to detect the standard library via macro __GLIBCXX__
+#include <vector>
+#endif
+#if MPT_COMPILER_MSVC || MPT_GCC_BEFORE(8,1,0) || MPT_CLANG_BEFORE(5,0,0) || (MPT_COMPILER_GCC && defined(__GLIBCXX__) && (defined(__MINGW32__) || defined(__MINGW64__))) || (MPT_COMPILER_CLANG && defined(__GLIBCXX__)) || (MPT_COMPILER_CLANG && MPT_OS_MACOSX_OR_IOS) || MPT_OS_OPENBSD || MPT_OS_EMSCRIPTEN || (defined(__clang__) && defined(_MSC_VER))
+#define MPT_COMPILER_QUIRK_NO_ALIGNEDALLOC
+#endif
+#endif
+
+
+
 // third-party library configuration
 
 #if defined(MODPLUG_TRACKER)
diff -ur common/CompilerDetect.h.bak common/CompilerDetect.h
--- common/CompilerDetect.h.bak	2019-07-26 12:11:44.000000000 +0200
+++ common/CompilerDetect.h	2019-08-24 20:27:59.937500000 +0200
@@ -311,9 +311,3 @@
 #define MPT_COMPILER_QUIRK_MSVC_STRINGSTREAM
 #endif
 
-#if MPT_CXX_AT_LEAST(17)
-#if MPT_COMPILER_MSVC || MPT_GCC_BEFORE(8,1,0) || MPT_CLANG_BEFORE(5,0,0) || (MPT_COMPILER_GCC && defined(__GLIBCXX__) && (defined(__MINGW32__) || defined(__MINGW64__))) || (MPT_COMPILER_CLANG && defined(__GLIBCXX__)) || (MPT_COMPILER_CLANG && MPT_OS_MACOSX_OR_IOS) || MPT_OS_OPENBSD || MPT_OS_EMSCRIPTEN || (defined(__clang__) && defined(_MSC_VER))
-#define MPT_COMPILER_QUIRK_NO_ALIGNEDALLOC
-#endif
-#endif
-
