diff --git a/Makefile.in b/Makefile.in
index 3a70a71..070f10d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -166,6 +166,15 @@ bin:
 lib:
 	$(MKDIR) $@
 
+install-static:	lib static
+	$(MKDIR) -p $(DESTDIR)$(INSTALL_INCDIR)
+	$(MKDIR) -p $(DESTDIR)$(INSTALL_LIBDIR)
+	$(MKDIR) -p $(DESTDIR)$(INSTALL_PKGDIR)
+	cp $(PUBLIC_INCLUDES) $(DESTDIR)$(INSTALL_INCDIR)
+	cp $(STATIC_TARGET) $(DESTDIR)$(INSTALL_LIBDIR)
+	sed "s,%PREFIX%,$(PREFIX),;s,%LIBRARY_LIBS%,$(LIBRARY_LIBS)," rubberband.pc.in \
+	  > $(DESTDIR)$(INSTALL_PKGDIR)/rubberband.pc
+
 install:	all
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_BINDIR)
 	$(MKDIR) -p $(DESTDIR)$(INSTALL_INCDIR)
diff --git a/rubberband.pc.in b/rubberband.pc.in
index ac96edd..ba1256c 100644
--- a/rubberband.pc.in
+++ b/rubberband.pc.in
@@ -5,6 +5,7 @@ includedir=${prefix}/include
 
 Name: rubberband
 Version: 1.8.2
-Description: 
+Description: An audio time-stretching and pitch-shifting library and utility program
 Libs: -L${libdir} -lrubberband
+Libs.private: %LIBRARY_LIBS%
 Cflags: -I${includedir} 
